net/dns/direct.go

```diff
@@ -50,6 +50,10 @@ func readResolv(r io.Reader) (config OSConfig, err error) {
 	scanner := bufio.NewScanner(r)
 	for scanner.Scan() {
 		line := strings.TrimSpace(scanner.Text())
+		i := strings.IndexByte(line, '#')
+		if i >= 0 {
+			line = line[:i]
+		}
 
 		if strings.HasPrefix(line, "nameserver") {
```

Comment:
so this matches "nameserver1.2.3.4" without a space, eh? :)

Reply:
picky picky.

done. (nice to have tests!)